flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationExecutor.java

```diff
@@ -324,15 +332,16 @@ public TableEnvironmentInternal getTableEnvironment() {
 
         final Executor executor =
                 lookupExecutor(streamExecEnv, sessionContext.getUserClassloader());
+        ResourceManager resourceManager = sessionContext.getSessionState().resourceManager.copy();
```

Comment:
Currently the ResourceManager uses lazy loading strategy to load the used resources. That is, the resource manager will add the function jar resources(*the function is not temporary*) when the query uses this. If we use a seperate resource manager here, it has some side efforts:

1. we will keep downloading from the remote function jar when using it
2. it's hard to track whether the resource manager contains the resource with the same resource uri.